src/core/event-handler.js

```diff
@@ -165,80 +148,31 @@ class EventHandler {
      * Fire an event, all additional arguments are passed on to the event listener.
      *
      * @param {string} name - Name of event to fire.
-     * @param {*} [arg1] - First argument that is passed to the event handler.
-     * @param {*} [arg2] - Second argument that is passed to the event handler.
-     * @param {*} [arg3] - Third argument that is passed to the event handler.
-     * @param {*} [arg4] - Fourth argument that is passed to the event handler.
-     * @param {*} [arg5] - Fifth argument that is passed to the event handler.
-     * @param {*} [arg6] - Sixth argument that is passed to the event handler.
-     * @param {*} [arg7] - Seventh argument that is passed to the event handler.
-     * @param {*} [arg8] - Eighth argument that is passed to the event handler.
+     * @param {...*} args - Arguments that are passed to the event handler.
      * @returns {EventHandler} Self for chaining.
      * @example
      * obj.fire('test', 'This is the message');
      */
-    fire(name, arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8) {
-        if (!name || !this._callbacks[name])
-            return this;
-
-        let callbacks;
-
-        if (!this._callbackActive[name]) {
-            this._callbackActive[name] = this._callbacks[name];
-        } else {
-            if (this._callbackActive[name] === this._callbacks[name])
-                this._callbackActive[name] = this._callbackActive[name].slice();
-
-            callbacks = this._callbacks[name].slice();
-        }
+    fire(name, ...args) {
```

Comment:
When this transpired into ES5, does it use `arguments`? If it does, then, unfortunately, this will lead to a lack of V8 optimization on this method, unless V8 team made a change here. If it still the case, then calling `fire` will lead to always unoptimized JS under V8, making it significantly slower in some cases.

Reply:
Yea, ES5 code looks like this:

```js
		_proto.fire = function fire(name) {
			if (!name || !this._callbacks.has(name)) return this;
			var handlers = this._callbacks.get(name);
			if (!handlers) return this;
			for (var _len = arguments.length, args = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
				args[_key - 1] = arguments[_key];
			}
			for (var i = 0; i < handlers.length; i++) {
				var handler = handlers[i];
				handler.callback.apply(handler.scope, args);
				if (handler.once) {
					handlers.splice(i, 1);
					i--;
				}
			}
			if (handlers.length === 0) {
				this._callbacks.delete(name);
			}
			return this;
		};
```

I didn't profile it, but I trust your intuition here, the extra `for`-loop/`Array` creation doesn't look so appealing either.